examples/random-login/main_test.go

```diff
@@ -185,3 +185,50 @@ func TestRandomSalt(t *testing.T) {
 		},
 	}.Run(t, server)
 }
+
+func TestRandomLogin(t *testing.T) {
+	server, err := integration.NewServer(t.Context(),
+		"random-login",
+		semver.Version{Minor: 1},
+		integration.WithProvider(provider()),
+		integration.WithMocks(&integration.MockResourceMonitor{
+			NewResourceF: func(args integration.MockResourceArgs) (string, property.Map, error) {
+				// mock the registration of the component's resources
+				switch {
+				case args.TypeToken == "random:index/randomId:RandomId" && args.Name == "login-id":
+					assert.Equal(t, 8.0, args.Inputs.Get("byteLength").AsNumber())
+					return "user", property.Map{}, nil
+
+				case args.TypeToken == "random:index/randomInteger:RandomInteger" && args.Name == "login-length":
+					assert.Equal(t, 8.0, args.Inputs.Get("min").AsNumber())
```

Comment:
Nit (need not be addressed): I prefer not using `AsNumber()` in this scenario, since it can panic. This is equivalent:
```suggestion
					assert.Equal(t, property.New(8.0), args.Inputs.Get("min"))
```

Reply:
Yeah that does look better. One little surprise upon trying this style is that it would assert over all components of the value, e.g. the deps.  For better/worse it led to:

```
        	Error:      	Not equal: 
        	            	expected: property.New(12.0)
        	            	actual  : property.New(12.0).WithDependencies([]urn.URN{"urn:pulumi:stack::project::random-login:index:RandomLogin$random:index/randomInteger:RandomInteger::login-length"})
```